knot-thing/src/main/java/com/cesar/knot_sdk/knot_data/KNoTConfig.kt

```diff
@@ -0,0 +1,19 @@
+package com.cesar.knot_sdk.knot_data
+
+/**
+ * This class holds the definitions of periodicity and of upper and lower limits for
+ * the associated KNoTValue.
+ *
+ * @property eventFlags theses are the flags that
+ * @property timeSec this is a timeout for the update of the KNoTValue
+ * @property lowerLimit this is the inferior limit of acceptable KNoTValues
+ * @property upperLimit this is the superior limit of acceptable KNoTValues
+ */
+data class KNoTConfig<T>(
+    val lowerLimit : KNoTValue<T>,
+    val upperLimit : KNoTValue<T>,
+    val timeSec : Int = Int.MAX_VALUE,
+    val lowerThreshold : Boolean = false,
```

Comment:
I do not think it would be inconsistent with the KNoT Protocol since its characteristics are kept. If you will keep the flags you should add a time flag.